### ISSUE TITLE
Theming issue on Ubuntu 21.04 with Wayland session

### ISSUE THREAD

**Issue by tunix** [tunix/alacritty-snap#16]
After upgrading to 21.04 and Wayland, alacritty started with default window decorations & cursor theme. I'm investigating this issue and also created a [forum topic](https://forum.snapcraft.io/t/wayland-decoration-issues-for-alacritty/24528).

![alacritty_wayland](https://user-images.githubusercontent.com/47731/119121746-75a9e180-ba36-11eb-80f9-a4ae82c5384d.png)

**Comment by elcste** [comment]
Not sure if it's good news or bad news, but I don't think these issues are specific to the Snap. I just tried this snap and a deb package on a mostly-default Ubuntu 21.04 install with GNOME/Wayland, and both issues show for both versions.

For the decoration, Wayland GNOME only supports CSDs, and this is Alacritty's correct decoration. Since it's not a GTK app, it doesn't have a GTK headerbar that fits the system theme. I think you can disable decorations in the Alacritty config if you just want a bare window.

For the cursor, I have seen it follow the theme under Sway with a deb package version, but noticed the same with GNOME/Wayland. I can't remember if I had something extra to do or if it just worked, and I don't have a machine with Sway installed and set up right now.

**Comment by tunix** [comment]
@elcste - So you're saying that, the window decorations and cursor theme are the same for both the snap & deb package, is that right? That's unfortunate. 😞 

In my experience, Telegram used its non-GTK decorations once you open it in a Wayland session on previous versions of Ubuntu. (pre-21.04) I now use Ubuntu 21.04 with Wayland and I don't see that. It rather uses GTK decorations. (I have `Use system window frame` setting enabled) And the cursor is also aligns with my theme setting.

**Comment by elcste** [comment]
@tunix Correct, the deb and the snap both show these decorations and that white cursor theme in Ubuntu GNOME/Wayland for me.

I previously only used GNOME/Wayland very briefly, but I remember now seeing it look like this in the past also with a debian package. Under Sway, I only saw this decoration, but usually used it with the decorations disabled. The cursor theme worked for me under Sway, but again I can't remember if it just worked or if I had to do something.

I've been playing around with Kubuntu, and I just tried this snap. Under Plasma/Wayland, the decoration *does* match the usual Plasma decoration and the cursor follows the system theme, so it seems it is possible under Wayland. Kwin supports SSD, though, so I still think Alacritty would have to add a GTK header bar to match the system theme under GNOME, but I could be wrong. The incorrect cursor theme seems to be a GNOME/Wayland and Alacritty issue.
